assembly/src/assembler/instruction/u32_ops.rs

```diff
@@ -399,40 +310,31 @@ fn handle_arithmetic_operation(
     imm: Option<u32>,
 ) -> Result<Option<CodeBlock>, AssemblyError> {
     let mut drop_high_bits = false;
-    let mut assert_u32_res = false;
 
     if let Some(imm) = imm {
         push_u32_value(span, imm);
     }
 
     match op_mode {
-        U32OpMode::Checked => {
-            span.push_op(U32assert2(ZERO));
-            assert_u32_res = true;
-        }
         U32OpMode::Wrapping => {
             drop_high_bits = true;
         }
         U32OpMode::Overflowing => {}
-        _ => unreachable!("unsupported operation mode"),
     }
 
     span.push_op(op);
 
-    if assert_u32_res {
-        span.add_ops([Eqz, Assert(ZERO)])
-    } else if drop_high_bits {
+    if drop_high_bits {
         span.add_op(Drop)
     } else {
         Ok(None)
     }
 }
```

Comment:
Could we not do something like this here?
```Rust
// in the wrapping mode, drop high 32 bits
if matches!(op_mode, U32OpMode::Wrapping) {
    span.add_op(Drop)
} else {
    Ok(None)
}
```
This could eliminate some of the code above.